### PR TITLE
Ruby Tic Tac Toe - Milestone 3

### PR DESCRIPTION
**Pull request by jmontanholi**
## In this branch we implemented the following:
* Transferred parts of logic that were in main to game_logic.rb
* Win condition, loops, and message.

### CODE REVIEW

File: lib/game_logic.rb
Comment (MkrtichSargsyan) [inline lib/game_logic.rb]: No puts/print and gets.chomp outside main.rb file.

File: lib/game_logic.rb
Comment (MkrtichSargsyan) [inline lib/game_logic.rb]: No puts/print and gets.chomp outside main.rb file. Instead of using puts in the design_board function 
- [ ]  Create a return string variable
- [ ]  Assign to that variable the board and return

**Review by MkrtichSargsyan — CHANGES_REQUESTED** [review]
## _Status:_ Changes Required :recycle:

Hi Guys. You've been doing a great job

Your project certainly is working fine. :100:
However, there are some changes that you need to make in order to get your project approved

  - [x]   Make development as default branch
  - [x]   No puts/print and gets.chomp outside main.rb file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
   - [x] The project should use at least two classes (Usually it uses three: Game, Board and Player). The code of the project should be logically divided between classes. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)
   - [x] Instead of ```attr_accessor :name``` use ```attr_reader :name``` 
   
Use attr_accessors if you really need it. Be aware that by using it you allow writing that property from outside the class. That variable can be set to anything. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
 



### Please make all the required changes before you submit another code review.
## Happy Coding :star: 

> ps: . If you have any questions about my review or need help, you can message me on slack ( @Mkrtich Sargsyan )

File: bin/main
Comment (MkrtichSargsyan) [inline bin/main:5]: Instead of ```attr_accessor :name``` use ```attr_reader :name```

File: lib/game_logic.rb
Comment (MkrtichSargsyan) [inline lib/game_logic.rb]: Instead of ```attr_accessor :name``` use ```attr_reader :name```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
# Status: Change Requested image

Amazing job so far guys @jmontanholi and @pountzas 👏👏

### HIghlights:
- User interface looks amazing it shows that you took your time to design it
- All combinations(horizontal, vertical, diagonal) are showing  a winner when elements are equal
- User input is validated, Kudos on that 

## To improve

- Add classes in different files so that your code looks neat like Game and Board classes they can be in src folder in separate files
- Try to be dry like in this method
rev
try not to repeat yourself it makes your code neat and easy to read

### Work on that and submit for another code review

All the best,

Lydia Ingabire

File: bin/main
Comment (henatan99) [inline bin/main:86]: - The game should display the board at winning condition before it returns a win message. 
Screenshot from 2021-03-27 02-54-04

File: bin/main
Comment (henatan99) [inline bin/main]: Class definition is prohibited in the project's requirements. Please move this class definition to another file.

**Review by henatan99 — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
Hi @jmontanholi, you have done a great job so far :+1: but there are few changes you need to make before your project gets approved. 

## To highlight 
- The game behaves as it is supposed to be 
- Simple code which does the job well 
## To Improve 
- When a player wins the game should display the board at winning condition before it outputs a win message (check the inline comment) 
- You cannot use any class definition in the main file. It is supposed to be a clean UI. As is suggested by the previous review, you can assign a variable to hold the board string when you define the Board class (you don't necessarily need puts in the Board class methods). 

* [x] The project uses at least two classes (Usually it uses three: Game, Board and Player). The code of the project should be logically divided between classes. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [x] All game-logic related resources (classes and/or modules) are kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

You can consider an optional comment below (not required for project submission) 
[OPTIONAL]
- You may give an option to players either to play again or not after a win or tie condition is met.

### Please work on the required changes and re-submit for another review. If you have any query, please find me via:
- [LinkedIn](https://www.linkedin.com/in/henok-mekonnen-2a251613/)
- [Twitter](https://twitter.com/henatan99)
- or Henok Mekonnen on Slack

**Review by henatan99 — APPROVED** [review]
# Project Approved :100: 
## Congratulations @jmontanholi and @pountzas, you did a very nice job. :fireworks: :champagne: 

- Thank you for implementing the requested changes 
- Your work meets the project requirements 

[OPTIONAL] comment 
- You may do some refactoring of your code to make it more elegant.

### You can merge the PR and submit the mandatory project completion form 

## Good Luck  :+1: 

In case you want to reach out to me use the below links:
- [LinkedIn](https://www.linkedin.com/in/henok-mekonnen-2a251613/)
- [Twitter](https://twitter.com/henatan99)
- Henok Mekonnen on Slack

**Pull request merged**
